### DISCUSSION TITLE
Is there a maximum to how long a string can be?

### DISCUSSION THREAD

**Discussion by sophief10** [jspsych/jsPsych#2333] [Q&A]
Hi all!

I am trying to run the following code:

```
var postPractice_partOne = {
            type: 'instructions',
            show_clickable_nav: true,
            allow_keys: false,
            pages: function() {
                return [`<p>Good job! Now we can move on to the real experimental trials.</p><p>Remember, you should indicate your choice using the left and right arrow keys:</p><p><strong><u>${arrow1} = Yes</u></strong>; you prefer the proposal to the default.</p><p><strong><u>${arrow2} = No</u></strong>; you prefer the default to the proposal.</p><p>At the end of the experiment, we will select <em>one</em> trial randomly from all the trials in the experimental session and pay you according to your choice on this trial. The choices you make during the experiment are indicated in "experimental dollars." When we pay you at the end of the study, these experimental dollars will be converted into actual physical money using an exchange rate that we will inform you about, and will lead to payments <u>on average</u> of $5 - $15, though the exact amounts will depend on your choices and the precise exchange rate.</p><p>Trials from the practice rounds will not be considered. Because only one trial will count, you should treat every trial as if it could be the one and only trial that determines how much you and your partner receive. Your decisions on other trials cannot and should not affect what you decide to do on the current trial.</p><p>Press "Next" when you are ready to begin.<p>`]
            }
        };
        finalTimeline.push(postPractice_partOne);
```

I am writing the code in VSCode, which highlights strings in red. However, for some reason, the red highlighting ceases once my string reaches a certain length:
![20211117_133305](https://user-images.githubusercontent.com/90805854/142261698-aba52d29-63c9-41ec-be45-f7212aa55a30.jpg)
![20211117_133322](https://user-images.githubusercontent.com/90805854/142261704-4da38f00-866f-4331-abcf-c0e465d49d0d.jpg)

My code is not working anyway, and I suspect that the failure has something to do with the highlighting problem.

Has anyone encountered this issue before? If so, how did you resolve it?

**Comment by jodeleeuw** [comment]
Hi @woodenchair,

If there is a limit on string length, it is much larger than the string you are using. 

Have you tried seeing if the error goes away if you use a shorter string here? I suspect that it's not the length of the string causing the problem. It could be that there is some formatting error in the string, but I don't see any. I'm guessing the problem lies somewhere else in the code.

**Reply by sophief10** [reply]
@jodeleeuw My code appears to be working now, mysteriously (though the highlighting still does not appear). I do not think I made any significant changes since I last tried it out. Perhaps the issues lies in the VSCode program.

Thank you for your help!

**Reply by kburkin** [reply]
Hi @woodenchair, it looks like the quote you used is the backtick not an apostrophe. If you find it not working again, try retyping your quotes. :)

**Reply by sophief10** [reply]
Thanks @kburkin, I will try it out with apostrophes and see if it works!
